### PR TITLE
[Document Intelligence] Generate for API 2024-07-31-preview

### PR DESCRIPTION
**Pull request by YalinLi0312**
This PR includes code changes generated from typespec def: https://github.com/Azure/azure-rest-api-specs-pr/pull/17637 , and updates for the breaking changes on existing tests and samples.

### CODE REVIEW

**Comment by kashifkhan** [comment]
can you also update the changelog to include the breaking changes please :)

File: sdk/documentintelligence/azure-ai-documentintelligence/CHANGELOG.md [hunk 1]
```diff
@@ -3,12 +3,40 @@
...
+- Added properties `model_id`, `confidence_threshold`, `features`, `query_fields` and `max_documents_to_analyze` to model `DocumentTypeDetails`.
+- Added property `allow_overwrite` to model `BuildDocumentClassifierRequest`.
+- Added properties `allow_overwrite` and `max_training_hours` to model `BuildDocumentModelRequest`.
+- Added properties `classifier_id`, `split` and `doc_types` to model `ComposeDocumentModelRequest`.
 
 ### Breaking Changes
-
-### Bugs Fixed
+- Removed support for extracting lists from analyzed documents:
+    - Removed models `DocumentList` and `DocumentListItem`.
```
Comment (kashifkhan) [inline sdk/documentintelligence/azure-ai-documentintelligence/CHANGELOG.md:28]: just checking - are there any replacements that should be used instead or these are simply removed and is no longer available?

File: sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_operations.py [hunk 2]
```diff
@@ -3740,1084 +2832,1323 @@ def _classify_document_initial(  # pylint: disable=inconsistent-return-statement
...
 
-        :param classifier_id: Unique document classifier name. Required.
-        :type classifier_id: str
-        :param classify_request: Classify request parameters. Required.
-        :type classify_request: ~azure.ai.documentintelligence.models.ClassifyDocumentRequest
+        :param model_id: Unique document model name. Required.
+        :type model_id: str
+        :param analyze_batch_request: Analyze batch request parameters. Default value is None.
+        :type analyze_batch_request: ~azure.ai.documentintelligence.models.AnalyzeBatchDocumentsRequest
+        :keyword pages: List of 1-based page numbers to analyze.  Ex. "1-3,5,7-9". Default value is
```
Comment (kashifkhan) [inline sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_operations.py:2864]:
```suggestion
        :keyword pages: Range of 1-based page numbers to analyze.  Ex. "1-3,5,7-9". Default value is
```
just a nit as pages doesnt take a `list`.

File: sdk/documentintelligence/azure-ai-documentintelligence/tests/test_dmac_copy_model.py [hunk 3]
```diff
@@ -67,13 +67,12 @@ def test_copy_model_successful(self, client, documentintelligence_storage_contai
 
         assert copy.api_version == model.api_version
         assert copy.model_id != model.model_id
+        assert copy.model_id == copy_auth["targetModelId"]
         assert copy.description == model.description
-        assert copy.created_date_time == model.created_date_time
-        assert copy.expiration_date_time == model.expiration_date_time
```
Comment (kristapratico) [inline sdk/documentintelligence/azure-ai-documentintelligence/tests/test_dmac_copy_model.py:72]: Were these properties removed or are we just not testing them anymore?

File: sdk/documentintelligence/azure-ai-documentintelligence/CHANGELOG.md [hunk 4]
```diff
@@ -3,12 +3,40 @@
 ## 1.0.0b4 (Unreleased)
 
 ### Features Added
+- Added support for the Analyze Batch Documents API:
```
Comment (kristapratico) [inline sdk/documentintelligence/azure-ai-documentintelligence/CHANGELOG.md:6]: Do you plan to add tests/samples for some of these new features?

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/async_samples/sample_manage_models_async.py [hunk 5]
```diff
@@ -62,12 +62,13 @@ async def sample_manage_models():
             print("Doc types the model can recognize:")
             for name, doc_type in model.doc_types.items():
                 print(f"Doc Type: '{name}' built with '{doc_type.build_mode}' mode which has the following fields:")
-                for field_name, field in doc_type.field_schema.items():
-                    if doc_type.field_confidence:
-                        print(
-                            f"Field: '{field_name}' has type '{field['type']}' and confidence score "
-                            f"{doc_type.field_confidence[field_name]}"
-                        )
+                if doc_type.field_schema:
```
Comment (kristapratico) [inline sdk/documentintelligence/azure-ai-documentintelligence/samples/async_samples/sample_manage_models_async.py:65]: Did this become optional in this version?

File: sdk/documentintelligence/azure-ai-documentintelligence/CHANGELOG.md (hunk 1 above)
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/CHANGELOG.md:28]: These two models were used in AnalyzeResult property which are removed in new API.

File: sdk/documentintelligence/azure-ai-documentintelligence/tests/test_dmac_copy_model.py (hunk 3 above)
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/tests/test_dmac_copy_model.py:72]: They are not removed, just not testing. Also, the created time and expiration time in copy are not always equivalent to the ones in original. So they're not necessary to check.

File: sdk/documentintelligence/azure-ai-documentintelligence/CHANGELOG.md (hunk 4 above)
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/CHANGELOG.md:6]: Yes, tests and samples for new features will be in a separate PR. Additionally, we need some customization for some new features, working on it now.

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/async_samples/sample_manage_models_async.py (hunk 5 above)
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/async_samples/sample_manage_models_async.py:65]: Yes. https://github.com/Azure/azure-sdk-for-python/blob/3f96dcf6270ffac8492b38b8d3bb338a0e39be54/sdk/documentintelligence/azure-ai-documentintelligence/CHANGELOG.md?plain=1#L36

File: sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_operations.py (hunk 2 above)
Comment (YalinLi0312) [reply sdk/documentintelligence/azure-ai-documentintelligence/azure/ai/documentintelligence/_operations/_operations.py]: Updated across this package. Will request the change in tsp.

File: sdk/documentintelligence/azure-ai-documentintelligence/samples/async_samples/sample_manage_models_async.py (hunk 5 above)
Comment (kristapratico) [reply sdk/documentintelligence/azure-ai-documentintelligence/samples/async_samples/sample_manage_models_async.py:65]: Ah I missed that, thanks for point it out!

**Review by kristapratico — APPROVED** [review]

**Review by kashifkhan — APPROVED** [review]

**Pull request merged**